sdk/storage/azfile/internal/generated/zz_directory_client.go

```diff
@@ -0,0 +1,744 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+// DO NOT EDIT.
+
+package generated
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// DirectoryClient contains the methods for the Directory group.
+// Don't use this type directly, use NewDirectoryClient() instead.
+type DirectoryClient struct {
+	endpoint string
+	pl       runtime.Pipeline
+}
+
+// NewDirectoryClient creates a new instance of DirectoryClient with the specified values.
+// endpoint - The URL of the service account, share, directory or file that is the target of the desired operation.
+// pl - the pipeline used for sending requests and handling responses.
+func NewDirectoryClient(endpoint string, pl runtime.Pipeline) *DirectoryClient {
+	client := &DirectoryClient{
+		endpoint: endpoint,
+		pl:       pl,
+	}
+	return client
+}
+
+// Create - Creates a new directory under the specified share or parent directory.
+// If the operation fails it returns an *azcore.ResponseError type.
+// Generated from API version 2020-10-02
+// fileAttributes - If specified, the provided file attributes shall be set. Default value: ‘Archive’ for file and ‘Directory’
```

Comment:
Parameters with default values should be optional.  Can you check if these are marked as required?

Reply:
These are required parameters as per,
https://learn.microsoft.com/en-us/rest/api/storageservices/create-directory#request-headers